libcxx/test/support/checking_mutex.h

```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
```

Comment:
Can you please add a basic test that ensures that `std::unique_lock<std::mutex>` works properly? We used to test that implicitly via other tests, but that's not the case anymore so we should have something that ensures basic interoperability.